Debugging

orlichter1 opened the pull request:
## Status
Ready

## Description
Moving things from internal conetent

orlichter1 commented:
@anara123 Added the links in the readme

kirbles19 commented:
@orlichter1 Done.

anara123 commented:
@orlichter1 the build fails. I think it is because you have directory with space in it. change that directory name

orlichter1 commented:
@anara123 Hey I added "[Design](https://github.com/demisto/content/tree/Internal-public/docs/integration_design)" (and renamed it as "IntegrationDesign") and "[Playbook Conventions](https://github.com/demisto/content/tree/Internal-public/docs/playbook_conventions)". Could you have a look and see that it fits what you wanted?
p.s. I already updated the links to look in master branch so they might not work in this one.

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+
+## Why do I need a design document in the first place?
+You might be thinking "Why do I need to write this document? Why not just start writing?" 
+Well, we beleive that writing such a document before starting any code writing will help you better understand your use of this integration, understand what exactly you want to do and what you want to get out of each command or fetch.
```
```suggestion
Well, we believe that writing such a document before starting any code writing will help you better understand your use of this integration, understand what exactly you want to do and what you want to get out of each command or fetch.
```

anara123 commented on docs/integration_design/README.MD:
````diff
@@ -0,0 +1,104 @@
+```
+## Parameters
+In this section write what are the parameters you need in order to create an instance of the product and connect to it. These parameters will be accessible and used in all the integration. 
+We require 2 fields as part of the parameters:
````
mention common parameters that we see in integrations, like `url`, `apikey` / `apitoken`/ `credentials (username, password)`, if the integrations fetches incidents then `query` which will query events/incidents/tickets

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+2. Command arguments
+3. Command outputs
+#### Command Name
+We have a standard for a command's name: `productName-command`
```
```suggestion
We have a standard for a command's name: `brand-action`
```

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+If your commands checks the maliciousness of an IP/Domain/URL/File name your command simply `ip`\ `domain`\ `url`\ `file`. If this is the case make sure your outputs (which we'll get to in a second) match the [context standards](https://github.com/demisto/content/tree/master/docs/context_standards).
+
+#### Command argument
+These are the arguments for the command for example `!intsights-get-alert-activities alert-id=`
```
```suggestion
These are the arguments for the command for example `!intsights-get-alert-activities alert-id=`, `!ip ip="8.8.8.8"`
```

anara123 replied:
argument names must be snake case. 
good: `alert-id`.
bad: `alertId`

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+
+The Context Outupt will be a JSON in the following format `{Product.TypeofOuput.Info}`. For more info about this have a look [here](https://github.com/demisto/content/tree/master/docs/context_and_ouputs)
+
+### Command Deisgn
```
```suggestion
### Command Design
```

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+###### Arguments
+argument | type | description|mandatory
+---|---|---|---
+InterfaceName|String|the name of the interface|no
```
InterfaceName is not according to our standard.

anara123 commented on docs/integration_design/README.MD:
```diff
@@ -0,0 +1,104 @@
+
+RuleID|InterfaceName|SourceIP|DestIP|Position|
+---|---|---|---|---|
+...|...|...|...|...|
```
present some data here

orlichter1 commented:
@anara123 Added the Command examples and changed your requests for Integration Design :)

orlichter1 commented:
@amshamah419 could you please go over [this](https://github.com/demisto/content/edit/Internal-public/docs/integration_design/README.MD?pr=%2Fdemisto%2Fcontent%2Fpull%2F4608)

amshamah419 commented:
Not quite done yet, will finish tomorrow.

amshamah419 commented:
Looks good @anara123 @orlichter1
